src/software/sensor_fusion/sensor_fusion.cpp

```diff
@@ -203,17 +207,79 @@ void SensorFusion::updateWorld(const SSL_DetectionFrame &ssl_detection_frame)
             friendly_team = createFriendlyTeam(blue_team);
             enemy_team    = createEnemyTeam(yellow_team);
         }
+        updateRobotStatesMap();
     }
 
     if (new_ball_state)
     {
-        if (ball)
+        updateBall(*new_ball_state);
+    }
+}
+
+void SensorFusion::updateBall(TimestampedBallState new_ball_state)
+{
+    if (!ball_states.empty() &&
+        new_ball_state.timestamp() < ball_states.front().timestamp())
+    {
+        throw std::invalid_argument(
+            "Error: Trying to update ball state using a state older then the current state");
+    }
+
+    ball_states.push_front(new_ball_state);
+
+    if (ball)
+    {
+        ball->updateState(new_ball_state);
+    }
+    else
+    {
+        ball = Ball(new_ball_state);
+    }
+}
+
+void SensorFusion::updateRobotStatesMap()
+{
+    for (const auto &robot : friendly_team.getAllRobots())
```

Comment:
helper w/ team and robot state map as arguments will allow us to not repeat the same code, could be a private helper of sensor fusion?

Reply:
good point

Reply:
cleaned up